src/Stack.java

```diff
@@ -0,0 +1,50 @@
+// A stack class for characters.  
+class Stack {
+	private char stck[]; // this array holds the stack
+	private int tos; // top of stack
+
+	// Construct an empty Stack given its size.
+	Stack(int size) {
+		stck = new char[size]; // allocate memory for stack
+		tos = 0;
+	}
+
+	// Construct a Stack from a Stack.
+	Stack(Stack ob) {
+		tos = ob.tos;
+		stck = new char[ob.stck.length];
+
+		// copy elements
+		for (int i = 0; i < tos; i++)
+			stck[i] = ob.stck[i];
+	}
+
+	// Construct a stack with initial values.
+	Stack(char a[]) throws StackFullException {
```

Comment:
Do you need to specify that the constructor throws the exception, since it's thrown by the push() method?